### ISSUE TITLE
Joint motor stability

### ISSUE THREAD

**Issue by emilk** [dimforge/rapier#125]
I've been trying out [the new joint motors](https://github.com/dimforge/rapier/pull/119) (thanks for adding those!).

I've found a stability issue, which should be easy to resolve, but I wanted to make an issue before I jump to PR.

The issue is that for high values of stiffness and damping the motor will overshoot its target, producing instabilities. Here is an excerpt from <https://github.com/dimforge/rapier/blob/master/src/dynamics/solver/joint_constraint/revolute_velocity_constraint.rs#L113:L121>:

``` rust
        if stiffness != 0.0 {
            motor_angle = joint.estimate_motor_angle(&rb1.position, &rb2.position);
            motor_rhs += (motor_angle - joint.motor_target_pos) * stiffness;
        }

        if damping != 0.0 {
            let curr_vel = rb2.angvel.dot(&motor_axis2) - rb1.angvel.dot(&motor_axis1);
            motor_rhs += (curr_vel - joint.motor_target_vel) * damping;
        }
```

I think the `stiffness` here (which comes from `combine_coefficients`) should be clamped to the `[0, 1/dt]` interval, and `damping` should be clamped to `[0, 1]` (or possibly to `[0, velocity_based_erp_inv_dt]` and `[0, velocity_solve_fraction]` respectively once https://github.com/dimforge/rapier/pull/116 is merged). This should prevent overshoot while allowing the motor constraint to satisfy as quickly as possible. With such clamping the user can specify an infinite stiffness and damping and have it mean "as quickly as any other constraint"

**Comment by sebcrozet** [comment]
Hi! As you suggest in #126, I feel like we should not clamp the value and simply document what the recommended values are. This would avoid some needless additional complication in our code, and this clamping could prevent some use-cases where, for example, one would want to setup an overly stiff spring to demonstrate how it behaves (for educational purposes).

**Comment by sebcrozet** [comment]
The modeling of joints in Rapier has changed dramatically since this issue was open, and the suggested clamping is no longer relevant. I think we can close this and open a different issue if problems are found with the new implementation.
